src/serverjar/serverjar_paper.py

```diff
@@ -36,66 +40,69 @@ def getInstalledPaperVersion(localPaperName):
 
 def findVersionGroup(mcVersion):
     versionGroups = ['1.16', '1.15']
-    versionGroupFound = False
+    if mcVersion is None:
+        return False
     for versionGroup in versionGroups:
-        if versionGroupFound == True:
-            break
         url = f"https://papermc.io/api/v2/projects/paper/version_group/{versionGroup}/builds"
         papermcdetails = doAPIRequest(url)
         papermcVersionForMc = papermcdetails["versions"]
         for versions in papermcVersionForMc:
             if versions == mcVersion:
-                versionGroupFound = True
                 paperVersionGroup = versionGroup
-                break
+                return paperVersionGroup
             if versionGroup == mcVersion:
-                versionGroupFound = True
                 paperVersionGroup = versionGroup
-                break
-
-    return paperVersionGroup
+                return paperVersionGroup
+    return False  # Not found
 
 
 def findBuildVersion(wantedPaperBuild):
     versionGroups = ['1.16', '1.15']
-    paperBuildFound = False
+    if wantedPaperBuild is None:
+        return False
     for versionGroup in versionGroups:
-        if paperBuildFound is True:
-            break
         url = f"https://papermc.io/api/v2/projects/paper/version_group/{versionGroup}/builds"
         papermcdetails = doAPIRequest(url)
         paperMcBuilds = papermcdetails["builds"]
         for build in paperMcBuilds:
             paperBuild = str(build["build"])
             if paperBuild == wantedPaperBuild:
-                paperBuildFound = True
                 paperVersionGroup = build["version"]
-                break
-    return paperVersionGroup
-
+                return paperVersionGroup
+    return False  # Not found
 
 def findLatestBuild(paperVersionGroup):
+    if paperVersionGroup is None:
+        return False
     url = f"https://papermc.io/api/v2/projects/paper/version_group/{paperVersionGroup}/builds"
     papermcbuilds = doAPIRequest(url)
+    if papermcbuilds["status"] == 404:
```

Comment:
They will run in an key error because the site won't return a status code:
```
if papermcbuilds["status"] == 404:
KeyError: 'status'
```

If you delete the two lines 79 & 80 the code works without the check for a successfull pull.


```suggestion
```

Reply:
I have a different method to resolve this without entirely removing the check. It is currently in my working branch. I will push it soon.
```PY
if "status" in papermcbuilds:  # Checks if the API returns a status. This means that there was an error.
    return False
```